Speed up tests SolvatedPDBComponent (#751)
OpenFreeEnergy/gufe: news/speedup_tests_solpdbc.rst

**Added:**

* <news item>

**Changed:**

> * Speedup the counting of the number of water molecules. Tests for the SolvatedPDBComponent are now using a PDB test file with solvated ligands to speedup tests.

Reviewer: ```suggestion
* Sped up the counting of the number of water molecules in a SolvatedPDBComponent.
```
No need to include non-API changes!
